degirum_tools/line_count.py

```diff
@@ -52,6 +52,37 @@ def __init__(self):
         self.for_class: Dict[str, SingleLineCounts] = {}
 
 
+class SingleVectorCounts:
+    """Class to hold vector crossing counts"""
+
+    def __init__(self):
+        self.in_direction: int = 0
+        self.out_direction: int = 0
+
+    def __eq__(self, other):
+        if not isinstance(other, SingleVectorCounts):
+            return NotImplemented
+        return (
+            self.in_direction == other.in_direction
+            and self.out_direction == other.out_direction
+        )
+
+    def __iadd__(self, other):
+        if not isinstance(other, SingleVectorCounts):
+            return NotImplemented
+        self.in_direction += other.in_direction
+        self.out_direction += other.out_direction
+        return self
+
+
+class VectorCounts(SingleVectorCounts):
+    """Class to hold total vector crossing counts and counts for multiple classes"""
+
+    def __init__(self):
+        super().__init__()
+        self.for_class: Dict[str, SingleVectorCounts] = {}
+
+
 class LineCounter(ResultAnalyzerBase):
     """
     Class to count object tracking trails crossing lines.

```

Comment:
Need to significantly update this comment block. Explaining new functionality and new result attributes